crypt4gh/keys/__init__.py

```diff
@@ -155,6 +157,29 @@ def run(argv=sys.argv[1:]):
 
     pubkey = os.path.expanduser(args['--pk'])
     seckey = os.path.expanduser(args['--sk'])
+    comment = args['-C'].encode() if args['-C'] else None
+
+    if args['--reencrypt']:
+        seckeypath = os.path.expanduser(seckey)
+        if not os.path.exists(seckeypath):
+            raise ValueError('Secret key not found')
+        old_passphrase = os.getenv('C4GH_PASSPHRASE')
+        cb = lambda: old_passphrase if old_passphrase else partial(getpass, prompt=f'Passphrase for {seckey}: ')
+        passphrase1 = getpass(prompt=f'Enter new passphrase for {args["--sk"]} (empty for no passphrase): ').encode()
+        passphrase2 = getpass(prompt=f'Enter new passphrase for {args["--sk"]} (again): ').encode()
+        if passphrase1 != passphrase2:
+            print('Passphrases do not match', file=sys.stderr)
+            return 1
+        seckey_reencrypted = c4gh.encode_private_key(
+            key = get_private_key(seckeypath, cb),
```

Comment:
I think you should try to unlock the key earlier (whether it's locked or not), and if that passes, you can ask twice for the new passphrase.